kernel/smp.c

```diff
@@ -115,7 +115,7 @@ void z_smp_init(void)
 
 	unsigned int num_cpus = arch_num_cpus();
 
-	for (int i = 1; i < num_cpus; i++) {
+	for (unsigned int i = 1; i < num_cpus; i++) {
```

Comment:
Observation: Although this fixes the mixing of unsigned and int variables on this line, it introduces subsequent mixing a few lines down with start_cpu().